src/libsyntax/ext/expand.rs

```diff
@@ -1212,24 +1211,6 @@ fn mark_tts(tts: &[TokenTree], m: Mrk) -> Vec<TokenTree> {
     noop_fold_tts(tts, &mut Marker{mark:m, expn_id: None})
 }
 
-/// Check that there are no macro invocations left in the AST:
-pub fn check_for_macros(sess: &parse::ParseSess, krate: &ast::Crate) {
-    visit::walk_crate(&mut MacroExterminator{sess:sess}, krate);
-}
-
-/// A visitor that ensures that no macro invocations remain in an AST.
-struct MacroExterminator<'a>{
-    sess: &'a parse::ParseSess
-}
-
-impl<'a, 'v> Visitor<'v> for MacroExterminator<'a> {
-    fn visit_mac(&mut self, mac: &ast::Mac) {
-        self.sess.span_diagnostic.span_bug(mac.span,
-                                           "macro exterminator: expected AST \
-                                           with no macro invocations");
-    }
-}
-
```

Comment:
I don't think it's useful. If we accidentally leave a macro in the AST after expansion, the [folds](https://github.com/rust-lang/rust/blob/81aeb0cddea5a614a4eebc0f0384fa27a7eb011f/src/libsyntax/fold.rs#L173-L180) and [visits](https://github.com/rust-lang/rust/blob/81aeb0cddea5a614a4eebc0f0384fa27a7eb011f/src/libsyntax/visit.rs#L102-L109) already panic. `resolve` and `lowering` also panic appropriately on unexpanded macros.